File: UnitTest/gsd/demo/Module1/ModuleModule1.cs

using System;
using System.Threading.Tasks;
using Zeze.Net;
using Zeze.Util;

namespace demo.Module1
{
    public sealed partial class ModuleModule1 : AbstractModule
    {
#pragma warning disable CA1822 // Mark members as static
        public void Start(demo.App _)
#pragma warning restore CA1822 // Mark members as static
        {
        }

        public void Stop(demo.App _)
        {
        }

        protected override async Task<long> ProcessProtocol1(Protocol p)
        {
            var protocol = p as Protocol1;
            protocol.Send(protocol.Sender);
            Console.WriteLine(protocol.Argument);
            await new Rpc2().SendAsync(protocol.Sender); // not wait
            return ResultCode.Success;
        }

        protected override Task<long> ProcessProtocol3(Protocol p)
        {
            return Task.FromResult(ResultCode.NotImplement);
        }

        protected override Task<long> ProcessRpc1Request(Protocol p)
        {
            var rpc = p as Rpc1;
            rpc.SendResult();
            return Task.FromResult(ResultCode.Success);
        }

        protected override Task<long> ProcessProtocolNoProcedure(Protocol p)
        {
            throw new NotImplementedException();
        }

        public Table1 Table1 => _Table1;
        public Table2 Table2 => _Table2;
        public TableImportant TableImportant => _TableImportant;
        public tFlush Tflush => _tFlush;
    }
}

File: cxx/test/demo/Module1/ModuleModule1.cpp

// ZEZE_FILE_CHUNK {{{ IMPORT GEN
#include "ModuleModule1.h"
#include "Gen/demo/App.h"
// ZEZE_FILE_CHUNK }}} IMPORT GEN

namespace demo {
namespace Module1 {

    void ModuleModule1::Start() {
    }

    void ModuleModule1::Stop() {
    }

    int64_t ModuleModule1::ProcessProtocol3(Zeze::Net::Protocol* _p) {
        return Zeze::ResultCode::NotImplement;
    }

    int64_t ModuleModule1::ProcessProtocol4(Zeze::Net::Protocol* _p) {
        return Zeze::ResultCode::NotImplement;
    }

    int64_t ModuleModule1::ProcessRpc2Request(Zeze::Net::Protocol* _r) {
        return Zeze::ResultCode::NotImplement;
    }

    // ZEZE_FILE_CHUNK {{{ GEN MODULE
    ModuleModule1::ModuleModule1(demo::App* app) : AbstractModule(app) {
    }
    // ZEZE_FILE_CHUNK }}} GEN MODULE
}
}
